support-frontend/assets/pages/paper-subscription-checkout/components-checkout/summary.jsx

```diff
@@ -0,0 +1,263 @@
+// @flow
+
+import React, { Component } from 'react';
+import { hasDiscount, type Price, type Promotion } from 'helpers/productPrice/productPrices';
+import type { BillingPeriod } from 'helpers/billingPeriods';
+import styles from './summary.module.scss';
+import { PriceLabel } from 'components/priceLabel/priceLabel';
+import typeof GridImageType from 'components/gridImage/gridImage';
+import { type GridImg } from 'components/gridImage/gridImage';
+import SvgDropdownArrowUp from './dropdownArrowUp.svg';
+
+// Types
+
+export type DataListItem = {
+  title: string,
+  value: string,
+}
+
+type PropTypes = {|
+  billingPeriod: BillingPeriod,
+  changeSubscription: string,
+  dataList: DataListItem[],
+  description: ?string,
+  image: $Call<GridImageType, GridImg>,
+  productPrice: Price,
+  promotion: ?Promotion,
+  title: string,
+|};
+
+type StateTypes = {
+  showDropDown: boolean,
+}
+
+// Helpers
+
+type PropTypesDL = {|
+  dataList: DataListItem[],
+|};
+
+const DataList = ({ dataList }: PropTypesDL) => (
+  <div className={styles.dataList}>
+    {dataList.length > 0 &&
+      <dl className={styles.data}>
+        {dataList.map(item => [
+          <dt>{item.title}</dt>,
+          <dd>{item.value}</dd>,
+            ])
+        }
+      </dl>
+    }
+  </div>
+);
+
+type PropTypesPD = {|
```

Comment:
I think generally the convention in the code base is to declare all the types at the top of the file, it would be good to keep to this.